metadata-ingestion/src/datahub/ingestion/source/kafka_connect.py

```diff
@@ -879,6 +879,106 @@ def _extract_lineages(self):
         return
 
 
+@dataclass
+class SnowflakeSinkConnector:
+    connector_manifest: ConnectorManifest
+    report: KafkaConnectSourceReport
+
+    def __init__(
+        self, connector_manifest: ConnectorManifest, report: KafkaConnectSourceReport
+    ) -> None:
+        self.connector_manifest = connector_manifest
+        self.report = report
+        self._extract_lineages()
+
+    @dataclass
+    class SnowflakeParser:
```

Comment:
nested classes are generally an anti-pattern. is there a reason you're using them here?

Reply:
ok that's fien